### PR TITLE
fix: memory allocations

### PR DESCRIPTION
**Pull request by khevse**
Benchmarks 
```bash
go test -v -bench=. -benchmem -cpu 1
```

before
```bash
goos: darwin
goarch: amd64
pkg: github.com/go-ldap/ldap
BenchmarkFilterCompile            338554              3420 ns/op            1606 B/op         61 allocs/op
BenchmarkFilterDecompile         1578404               775 ns/op             155 B/op          8 allocs/op

BenchmarkFilterCompile            335392              3518 ns/op            1606 B/op         61 allocs/op
BenchmarkFilterDecompile         1557634               765 ns/op             155 B/op          8 allocs/op

BenchmarkFilterCompile            342820              3423 ns/op            1606 B/op         61 allocs/op
BenchmarkFilterDecompile         1535986               769 ns/op             155 B/op          8 allocs/op
```

after
```bash
goos: darwin
goarch: amd64
pkg: github.com/go-ldap/ldap
BenchmarkFilterCompile            622689              1914 ns/op            1581 B/op         36 allocs/op
BenchmarkFilterDecompile         1779561               671 ns/op             171 B/op          3 allocs/op

BenchmarkFilterCompile            616844              1932 ns/op            1581 B/op         36 allocs/op
BenchmarkFilterDecompile         1753670               683 ns/op             171 B/op          3 allocs/op

BenchmarkFilterCompile            635676              1927 ns/op            1581 B/op         36 allocs/op
BenchmarkFilterDecompile         1793448               676 ns/op             171 B/op          3 allocs/op
```

### CODE REVIEW

File: filter.go [hunk 1]
```diff
@@ -432,34 +437,46 @@ func compileFilter(filter string, pos int) (*ber.Packet, int, error) {
...
 
 // Convert from "ABC\xx\xx\xx" form to literal bytes for transport
-func escapedStringToEncodedBytes(escapedString string) (string, error) {
-	var buffer bytes.Buffer
-	i := 0
-	for i < len(escapedString) {
-		currentRune, currentWidth := utf8.DecodeRuneInString(escapedString[i:])
-		if currentRune == utf8.RuneError {
-			return "", NewError(ErrorFilterCompile, fmt.Errorf("ldap: error reading rune at position %d", i))
+func escapedStringToEncodedBytes(escapedString []byte) (string, error) {
```
Comment (johnweldon) [inline filter.go]: Might need to be renamed now... maybe just `encodeEscapedBytes`? Can't think of a better name off the top of my head.

**Comment by regeda** [comment]
@khevse please, copy files into `v3` folder as well.

**Comment by johnweldon** [comment]
> Copy all updated files into `v3` folder to respect Go modules.

Thanks for catching this.

**Comment by johnweldon** [comment]
> Copy all updated files into `v3` folder to respect Go modules.

Thanks for catching this.

**Comment by khevse** [comment]
Thank you!

**Pull request merged**